### ISSUE TITLE
Investigate using `<dialog>` `aria-modal` for sidebar 1.0

### ISSUE THREAD

**Issue by sparhami** [ampproject/amphtml#24382]
The current sidebar traps focus (both keyboard and screen reader) when it opens. This is done by adding `aria-hidden` and setting negative tab indicies on things outside of the dialog. Note that it does not use a focus trap that cycles back to the start, so that the correct behavior of focus leaving the browser window when swiping past the start/end is preserved.

The former case can be simplified using `aria-modal` and `role="dialog"` on the sidebar instead of adding `aria-hidden` to other elements. Support for various screen readers here is not exactly clear, but using this approach would simplify the code.

Another approach would be to wrap the contents of the sidebar using a `<dialog>` element. There are a few complications here for styling / animation, due to how the dialog element opens and positions itself, but this handles both screen readers and keyboard focus. This change cannot be done for the existing sidebar due to how things are structured and the way it can be customized, but is potentially something we could do for a new version of sidebar.

We would still need some polyfill behavior, as not all browsers support the dialog element.

/cc @leafsy @cathyxz
